### PR TITLE
improvement: Throw an error with a correct message

### PR DESCRIPTION
**Pull request by tgodzik**
Thanks to this the proper message will be displayed.

Fixes https://github.com/scalameta/metals-vscode/issues/1436

Screenshot from 2023-11-17 19-15-49

### CODE REVIEW

File: packages/metals-vscode/src/debugger/scalaDebugger.ts [hunk 1]
```diff
@@ -22,8 +22,9 @@ import { currentWorkspaceFolder } from "../util";
 const configurationType = "scala";
 
 export interface DebugSession {
-  name: string;
-  uri: string;
+  name: string | undefined;
+  uri: string | undefined;
+  error: string | undefined;
 }
```
Comment (kpodsiad) [inline packages/metals-vscode/src/debugger/scalaDebugger.ts:28]: are `{name: string, uri: string}` and `{error: string}` mutually exclusive?

File: packages/metals-vscode/src/debugger/scalaDebugger.ts [hunk 2]
```diff
@@ -191,13 +199,18 @@ class ScalaDebugServerFactory implements vscode.DebugAdapterDescriptorFactory {
         session.configuration
       );
 
-      if (debugSession === undefined) {
-        return null;
-      } else {
+      if (debugSession !== undefined && debugSession.uri !== undefined) {
```
Comment (kpodsiad) [inline packages/metals-vscode/src/debugger/scalaDebugger.ts:202]:
```suggestion
      if (debugSession?.uri) {
```

File: packages/metals-vscode/src/debugger/scalaDebugger.ts [hunk 3]
```diff
@@ -191,13 +199,18 @@ class ScalaDebugServerFactory implements vscode.DebugAdapterDescriptorFactory {
...
         return debugServerFromUri(debugSession.uri);
+      } else if (
+        debugSession !== undefined &&
+        debugSession.error !== undefined
+      ) {
```
Comment (kpodsiad) [inline packages/metals-vscode/src/debugger/scalaDebugger.ts:207]:
```suggestion
      } else if (debugSession?.error) {
```

File: packages/metals-vscode/src/debugger/scalaDebugger.ts [hunk 4]
```diff
@@ -191,13 +199,18 @@ class ScalaDebugServerFactory implements vscode.DebugAdapterDescriptorFactory {
...
+        throw new Error(debugSession.error);
+      } else {
+        throw new Error("Could not start debug adapter");
       }
```
Comment (kpodsiad) [inline packages/metals-vscode/src/debugger/scalaDebugger.ts:211]: `else { throw new Error("Could not start debug adapter") }` seems redundant, as there's fallback return at the end already

**Comment by tgodzik** [comment]
I realized, we can actually do it all on the server side by using lsp4j's exception:

https://github.com/scalameta/metals/pull/5864

**Pull request closed without merging**